docs/tutorials/csharp/stable-diffusion-csharp.md

```diff
@@ -0,0 +1,330 @@
+
+## Main Function
+The main function sets the prompt, number of inference steps, and the guidance scale. It then calls the `UNet.Inference` function to run the inference.
+
+The properties that need to be set are:
+ - `prompt` - The text prompt to use for the image
+ - `num_inference_steps` - The number of steps to run inference for
+ - `guidance_scale` - The scale for the classifier-free guidance
+ - `batch_size` - The number of images to create
+ - `height` - The height of the image. Default is 512 and must be a multiple of 8.
+ - `width` - The width of the image. Default is 512 and must be a multiple of 8.
+
+
+```csharp
+//Default args
+var prompt = "make a picture of green tree with flowers around it and a red sky";
+// Number of steps
+var num_inference_steps = 10;
+
+// Scale for classifier-free guidance
+var guidance_scale = 7.5;
+//num of images requested
+var batch_size = 1;
+// Load the tokenizer and text encoder to tokenize and encodethe text.
+var textTokenized = TextProcessing.TokenizeText(prompt);
+var textPromptEmbeddings = TextProcessing.TextEncode(textTokenized).ToArray();
+// Create uncond_input of blank tokens
+var uncondInputTokens = TextProcessing.CreateUncondInput();
+var uncondEmbedding = TextProcessing.TextEncode(uncondInputTokens).ToArray();
+// Concat textEmeddings and uncondEmbedding
+DenseTensor<float> textEmbeddings = new DenseTensor<float>(ne[] { 2, 77, 768 });
+for (var i = 0; i < textPromptEmbeddings.Length; i++)
+{
+    textEmbeddings[0, i / 768, i % 768] = uncondEmbedding[i];
+    textEmbeddings[1, i / 768, i % 768] = textPromptEmbeddings[i];
+}
+var height = 512;
+var width = 512;
+// Inference Stable Diff
+var image = UNet.Inference(num_inference_steps, textEmbeddings,guidance_scale, batch_size, height, width);
+// If image failed or was unsafe it will return null.
+if( image == null )
+{
+    Console.WriteLine("Unable to create image, please try again.");
+}
+```
+
+## Tokenization with ONNX Runtime Extensions
+
+The `TextProcessing` class has the functions to tokenize the text prompt and encoded it with the [CLIP model](https://huggingface.co/docs/transformers/model_doc/clip) text encoder.
+
+The CLIP tokenizer is not available in C# so in order to tokenize the text prompt we will use the [ONNX Runtime Extensions](https://github.com/microsoft/onnxruntime-extensions). The ONNX Runtime Extensions has a `custom_op_cliptok.onnx` file tokenizer that is used to tokenize the text prompt. The tokenizer is a simple tokenizer that splits the text into words and then converts the words into tokens. 
```

Comment:
"not available in C#" is true but felt like something was missing.
Consider "Instead of reimplementing the CLIP tokenizer in C#, we can leverage the cross-platform CLIP tokenizer implementation in ONNX Runtime Extensions."

Reply:
Fixed